### PR TITLE
Refactored patterns, added C#.

### PR DESCRIPTION
**Pull request by emeryberger**
See above.

### CODE REVIEW

File: src/cwhy/cwhy.py [hunk 1]
```diff
@@ -195,38 +215,20 @@ def __init__(self, args, diagnostic):
...
-                match = re.match(
-                    r'\s*File "(.*?)", line (\d+), in ([^\<].*)', diagnostic_lines[line]
-                )
-
-            line += 1
-
-            if not match:
-                continue
+        for (linenum, line) in enumerate(diagnostic_lines):
+            for (lang, pattern) in error_patterns:
```
Comment (nicovank) [inline src/cwhy/cwhy.py:226]: Maybe we should detect if we match on conflicting languages, can be done in the future if the issue comes up.

**Review by nicovank — APPROVED** [review]
Referencing https://github.com/plasma-umass/cwhy/issues/18. LGTM.

Do you have any C# code examples / reproduction commands?

**Comment by emeryberger** [comment]
For future reference, turns out you can install the C# compiler on the Mac by running `brew install mono` and then you invoke it as `mcs`, as in `mcs testme.cs`.

**Comment by emeryberger** [comment]
Added tests. Here's output for the C# error (correct):

----

```
% mcs testme.cs            
testme.cs(13,44): error CS1010: Newline in constant
testme.cs(15,12): error CS1525: Unexpected symbol `Console'
Compilation failed: 2 error(s), 0 warnings
% mcs testme.cs |& cwhy
The problem is that there is a syntax error in your code.
Specifically, on line 13, you have forgotten to close the string with
a quotation mark. The corrected line should be:

13             string message = "Hello World!!";

```

File: src/cwhy/cwhy.py (hunk 1 above)
Comment (emeryberger) [reply src/cwhy/cwhy.py:226]: True. Let's see what Go errors look like :).

Comment (emeryberger) [reply src/cwhy/cwhy.py:232]: Done, seems fine.

Comment (emeryberger) [reply src/cwhy/cwhy.py:232]: Resolving this for now.

**Pull request merged**